keras/layers/layer.py

```diff
@@ -1204,13 +1204,24 @@ def _track_variable(self, variable):
         if not self.trainable:
             variable.trainable = False
 
+        if backend.backend() == "torch" and hasattr(self, "torch_params"):
```

Comment:
Please move these 2 blocks to the torch Layer class. Just override the method, call super() and then add this logic. For both `_track_variable` and `_untrack_variable`. As a general rule we should strictly avoid backend-specific code in the main codebase.

Reply:
Done

Reply:
Fixed now.